cmd/stacker/build.go

```diff
@@ -4,9 +4,9 @@ import (
 	"fmt"
```

Comment:
Move all these under pkg/build.go ... etc instead of under cmd/stacker/...

Reply:
I don't quite understand how this would work. `build.go` and the other go files under `cmd/stacker` are part of the package `main`. I've tried moving all the go files except for main.go, but then I cannot import `pkg/stacker` because:
```
cmd/stacker/main.go:23:2: import "stackerbuild.io/stacker/pkg/stacker" is a program, not an importable package
```
Moving all of them under `pkg/stacker` doesn't seem to follow the go project layout, the main application should be under `cmd/stacker`, as described here: https://github.com/golang-standards/project-layout/tree/master/cmd

Reply:
yes, you have the change the package to "package stacker" on top line

Reply:
Thanks. So I should leave main.go in cmd/stacker and move the rest of the files under pkg/stacker?

Reply:
correct, and once files land there, change the "package main" to "package stacker"

Reply:
I've tried moving these files to `pkg/stacker` and changing `package main` to `package stacker`, but now files in `pkg/stacker` have a dependency on `cmd/stacker/main.go` (e.g. the `config` variable) and I'm not sure I can add `main.go` as a dependency to files in `pkg/stacker`, it definitely doesn't seem right to me.
Could you help me understand the reason why you want to move these files to `pkg/stacker`?

Reply:
For reference:
```
go build -tags "exclude_graphdriver_btrfs exclude_graphdriver_devicemapper containers_image_openpgp osusergo netgo " -ldflags "-X main.version=v0.30.1-9-g2ca5d76-dirty -X main.lxc_version=5.0.0~git2209-g5a7b9ce67 " -o stacker-dynamic ./cmd/stacker
# stackerbuild.io/stacker/pkg/stacker
pkg/stacker/build.go:94:25: undefined: config
pkg/stacker/build.go:101:25: undefined: shouldShowProgress
pkg/stacker/check.go:21:24: undefined: config
pkg/stacker/check.go:29:9: undefined: config
pkg/stacker/check.go:31:24: undefined: config
pkg/stacker/check.go:33:51: undefined: config
pkg/stacker/chroot.go:41:38: undefined: config
pkg/stacker/chroot.go:66:27: undefined: config
pkg/stacker/chroot.go:94:26: undefined: config
pkg/stacker/chroot.go:100:42: undefined: config
pkg/stacker/chroot.go:100:42: too many errors
make: *** [Makefile:30: stacker-dynamic] Error 2
```

Reply:
https://github.com/project-stacker/stacker/pull/356
^ try refactor based on this

Reply:
is it ok now?